### PR TITLE
Store transcripts in spaces - UI

### PR DESCRIPTION
**Pull request by albandum**
## Description

This PR enables the UI for storing transcripts in a Folder inside a Space. 
Since Spaces became available, it now makes sense to be able to store my meeting transcripts in a secure space if I want to. 

- We store the datasourceViewId 
- Separate toggles between 'storing transcripts' and 'processing transcripts' : I can want to have one, the other or both
- This is provider agnostic. This will enable us to add more providers to the 'meeting transcripts connector' like mojo, etc. 
- This does not import the history of calls
- When active, it only imports the meetings of the user that activated the feature.

This PR also edits the DatasourceViewSelector to pass the useCase all the way down, as in this selector I only want to be able to select **one** folder (the destination)
We're also only showing folders and not any other datasource types.

https://github.com/user-attachments/assets/8ecaf27c-8f33-4534-be19-b6e8c667b22a

## Deploy Plan

Run migration 108

### CODE REVIEW

File: front/migrations/db/migration_108.sql [hunk 1]
```diff
@@ -0,0 +1,3 @@
+-- Migration created on Oct 25, 2024
+ALTER TABLE "public"."labs_transcripts_configurations" ADD COLUMN "dataSourceViewId" INTEGER REFERENCES "data_source_views" ("id") ON DELETE SET NULL ON UPDATE CASCADE;
+ALTER TABLE "public"."labs_transcripts_configurations" DROP COLUMN "dataSourceId";
```
Comment (spolu) [inline front/migrations/db/migration_108.sql:3]: That can't be done in  one go can it?

Comment (spolu) [reply front/migrations/db/migration_108.sql:3]: Shouldn't there be a migration?

**Review by spolu — COMMENTED** [review]
Looks good, left one comment though

File: front/lib/resources/labs_transcripts_resource.ts [hunk 2]
```diff
@@ -131,18 +131,28 @@ export class LabsTranscriptsConfigurationResource extends BaseResource<LabsTrans
...
 
-    const dataSource = await DataSourceResource.fetchById(auth, dataSourceId);
+    if (dataSourceViewId === null) {
+      return this.update({ dataSourceViewId: null });
+    }
+
+    const dataSource = await DataSourceResource.fetchById(
+      auth,
+      dataSourceViewId
+    );
```
Comment (tdraier) [inline front/lib/resources/labs_transcripts_resource.ts:149]: Shouldn't it be DataSourceViewResource ?

File: front/pages/api/w/[wId]/labs/transcripts/[tId].ts [hunk 3]
```diff
@@ -121,8 +121,12 @@ async function handler(
         }
       }
 
-      if (dataSourceId !== undefined) {
-        await transcriptsConfiguration.setDataSourceId(auth, dataSourceId);
+      if (dataSourceViewId !== undefined) {
+        console.log("Setting data source id", dataSourceViewId);
```
Comment (tdraier) [inline front/pages/api/w/[wId]/labs/transcripts/[tId].ts:125]: ✂️

File: front/pages/w/[wId]/assistant/labs/transcripts/index.tsx [hunk 4]
```diff
@@ -246,26 +294,29 @@ export default function LabsTranscriptsIndex({
...
+        dataSourceView.dataSource.name;
+    } else {
+      successMessage = "The transcripts will not be stored.";
     }
+
     await makePatchRequest(
       transcriptConfigurationId,
       {
-        dataSourceId: dataSource ? dataSource.dataSource.sId : null,
+        dataSourceViewId: dataSourceView ? dataSourceView.dataSource.sId : null,
```
Comment (tdraier) [inline front/pages/w/[wId]/assistant/labs/transcripts/index.tsx:319]: `dataSourceView.dataSource.sId` is a datasource id, not a datasource view

**Review by tdraier — APPROVED** [review]

**Pull request merged**